rfcs/2020-10-29-3736-remap-chain-parsing.md

```diff
@@ -0,0 +1,181 @@
+# RFC 3736 - 2020-10-29 - Ability to chain parsing functions in the remap syntax
+
+Vector needs to be able to parse a variety of log formats in different ways. 
+
+This RFC proposes enhancements to the Remap language to give it the ability to chain multiple parsing strategies 
+in the language. For example, a number of regular expressions can be chained together 
+and each run sequentially until one passes.
+
+
+## Scope
+
+This RFC will focus on the syntax needed to enhance the Remap language to allow for multiple branches as well as the 
+functionality required to allow the scripts to most efficiently run within these branches.
+
+## Motivation
+
+The best way to achieve this currently is with the `swimlanes` transform.
+
+However, the complexity required to efficiently maintain multiple parsing strategies using swimlanes is fairly complex which
+can lead to error prone and inefficient transforms.
+
+See issues:
+
+[#2418](https://github.com/timberio/vector/issues/2418)
+[#1477](https://github.com/timberio/vector/issues/1477)
+
+## Doc-level Proposal
+
+Within a `remap` mapping it is possible to specify a number of conditions. Each condition will be run sequentially - the first condition 
+that passes the associated mapping block will be performed. This is performed using `if...else if...else`:
+
+```
+if ... {
+   ...
+} else if ... {
+   ... 
+} else if ... {
+   ... 
+} else {
+   ...
+}
+```
+
+The mapping will start at the condition specified by the first `if` and will proceed along each condition specified by each `else if` sequentially until
```

Comment:
minor formatting nit: it reads easier for reviewers if you limit line length to either 80 or 100 characters.